public/app/features/alerting/unified/rule-list/GroupedView.tsx

```diff
@@ -1,53 +1,44 @@
-import { css } from '@emotion/css';
-import { PropsWithChildren, ReactNode, useMemo } from 'react';
+import { useMemo } from 'react';
 import Skeleton from 'react-loading-skeleton';
 
-import { GrafanaTheme2 } from '@grafana/data';
-import { Button, Dropdown, Icon, IconButton, LinkButton, Menu, Stack, Text, useStyles2 } from '@grafana/ui';
-import { Trans } from 'app/core/internationalization';
-import { DataSourceNamespaceIdentifier, DataSourceRuleGroupIdentifier, RuleGroup } from 'app/types/unified-alerting';
-import { RulesSourceApplication } from 'app/types/unified-alerting-dto';
+import { Alert, ErrorBoundary, ErrorWithStack, Stack, Text } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import { ExternalRulesSourceIdentifier, RulesSourceIdentifier } from 'app/types/unified-alerting';
 
 import { featureDiscoveryApi } from '../api/featureDiscoveryApi';
-import { Spacer } from '../components/Spacer';
-import { WithReturnButton } from '../components/WithReturnButton';
-import { getDatasourceAPIUid, getExternalRulesSources } from '../utils/datasource';
-import { hashRule } from '../utils/rule-id';
+import { getExternalRulesSources, GrafanaRulesSource } from '../utils/datasource';
 
-import { AlertRuleLoader } from './AlertRuleLoader';
-import { ListGroup } from './components/ListGroup';
-import { ListSection } from './components/ListSection';
-import { DataSourceIcon } from './components/Namespace';
-import { LoadingIndicator } from './components/RuleGroup';
-import { usePaginatedPrometheusRuleNamespaces } from './hooks/usePaginatedPrometheusRuleNamespaces';
+import { PaginatedDataSourceLoader } from './PaginatedDataSourceLoader';
+import { PaginatedGrafanaLoader } from './PaginatedGrafanaLoader';
+import { DataSourceSection } from './components/DataSourceSection';
 
 const { useDiscoverDsFeaturesQuery } = featureDiscoveryApi;
-const GROUP_PAGE_SIZE = 40;
+
+export const GROUP_PAGE_SIZE = 40;
 
 export function GroupedView() {
   const externalRuleSources = useMemo(() => getExternalRulesSources(), []);
 
   return (
     <Stack direction="column" gap={1} role="list">
-      <GrafanaDataSourceLoader />
+      <DataSourceErrorBoundary rulesSourceIdentifier={GrafanaRulesSource}>
+        <PaginatedGrafanaLoader />
+      </DataSourceErrorBoundary>
       {externalRuleSources.map((ruleSource) => {
-        return <DataSourceLoader key={ruleSource.uid} uid={ruleSource.uid} name={ruleSource.name} />;
```

Comment:
nit: i would move the `DataSourceErrorBoundary/ PaginatedDataSourceLoader` wrappers here for data source rules in the same way we do with Grafana rules, for consistency.

Reply:
I moved `PaginatedDataSourceLoader` to a separate file